src/services/backendService.ts

```diff
@@ -144,7 +145,24 @@ export class BackendService {
         throw new MapiError("Unhandled", "Unable to complete change password request.");
     }
 
-    public async getDelegationUrl(action: DelegationAction, delegationParameters: {}): Promise<string> {
+    public async applyDelegation(action: DelegationAction, delegationParameters: Bag<string>): Promise<string> {
+        if (this.developerPortalType === DeveloperPortalType.managed) {
+            const queryParams = new URLSearchParams();
+            Object.keys(delegationParameters).map(key => {
+                const val = delegationParameters[key];
+                queryParams.append(key, val);
+            });
+            return `/${DelegationActionPath[action]}?${queryParams.toString()}`;
+        } else {
+            const delegationUrl = await this.getDelegationUrl(action, delegationParameters);
+            if (delegationUrl) {
+                location.assign(delegationUrl);
```

Comment:
I believe we can have a single place where we compose the delegation URL (I suggest in `getDelegationUrl` of `backendService`), call it from the runtime components and do redirects in them respectively (using router everywhere, instead of location.assign). Maybe even do encapsulation like this:

subscription.ts
const url = backendService.getSubscriptionDelegationUrl(subscriptionName)
router.navigateTo(url);

signin.ts:
const url = backendService.getSigninDelegationUrl(subscriptionName)
router.navigateTo(url);